python/cudf_polars/cudf_polars/experimental/repartition.py

```diff
@@ -0,0 +1,53 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+"""Repartitioning Logic."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from cudf_polars.dsl.ir import IR
+from cudf_polars.experimental.base import get_key_name
+from cudf_polars.experimental.dispatch import generate_ir_tasks
+from cudf_polars.experimental.utils import _concat
+
+if TYPE_CHECKING:
+    from collections.abc import MutableMapping
+
+    from cudf_polars.experimental.parallel import PartitionInfo
+    from cudf_polars.typing import Schema
+
+
+class Repartition(IR):
+    """Repartition a DataFrame."""
+
+    __slots__ = ()
+    _non_child = ("schema",)
+
+    def __init__(self, schema: Schema, df: IR):
+        self.schema = schema
+        self._non_child_args = ()
+        self.children = (df,)
```

Comment:
*question*: Right now we're using `Repartition` as a tag to mean "concatenate to a single partition". Should we already accept a `num_partitions` argument (which we raise if it's not `1` for)

Reply:
So far, we have avoided attaching explicit partition-count information to the IR objects themselves. Instead, we refer to the extracted PartitionInfo object to decide how many output partitions this is generating.